src/CommunityPatch/Patches/EquipDraggedNonCivilianItemPatch.cs

```diff
@@ -0,0 +1,296 @@
+        return false;
+      }
+      
+      if (SetIsCivilianItemMethodInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(SetIsCivilianItemMethodInfo));
+        return false;
+      }
+
+      if (RightItemListVmFieldInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(RightItemListVmFieldInfo));
+        return false;
+      }
+      
+      if (LeftItemListVmFieldInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(LeftItemListVmFieldInfo));
+        return false;
+      }
+
+      if (CurrentCharacterFieldInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(CurrentCharacterFieldInfo));
+        return false;
+      }
+      
+      if (UpdateCurrentCharacterIfPossiblePrefixMethodInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(UpdateCurrentCharacterIfPossiblePrefixMethodInfo));
+        return false;
+      }
+        
+      if (UpdateCurrentCharacterIfPossiblePostfixMethodInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(UpdateCurrentCharacterIfPossiblePostfixMethodInfo));
+        return false;
+      } 
+        
+      if (InitializeInventoryPostfixMethodInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(InitializeInventoryPostfixMethodInfo));
+        return false;
+      }
+      
+      if (AfterTransferPostfixMethodInfo == null) {
+        PrintMethodNotFoundByReflectionError(nameof(InitializeInventoryPostfixMethodInfo));
+        return false;
+      }
+
+      if (!IsTargetPatchable(UpdateCurrentCharacterIfPossibleMethodInfo, UpdateCurrentCharacterIfPossibleHashes)) {
+        return false;
+      }
+      
+      if (!IsTargetPatchable(InitializeInventoryMethodInfo, InitializeInventoryHashes)) {
+        return false;
+      }
+      
+      if (!IsTargetPatchable(AfterTransferMethodInfo, AfterTransferHashes)) {
+        return false;
+      }
+
+      return base.IsApplicable(game);
+    }
+
+    private static void MapLeftAndRightInventory(SPInventoryVM spInventoryVm, Action<SPItemVM> itemMapper) {
+      var rightItemListVm = (MBBindingList<SPItemVM>) RightItemListVmFieldInfo.GetValue(spInventoryVm);
+      var leftItemListVm = (MBBindingList<SPItemVM>) LeftItemListVmFieldInfo.GetValue(spInventoryVm);
+
+      if (rightItemListVm == null || leftItemListVm == null) {
+        return;
+      }
+      
+      leftItemListVm.ApplyActionOnAllItems(itemMapper);
+      rightItemListVm.ApplyActionOnAllItems(itemMapper);
+    }
+
+    private static void NonCivilianPerkItemsEquipableState(SPInventoryVM spInventoryVm, Func<SPItemVM, bool> isItemApplicableToPerk, bool state) {
+
+      Action<SPItemVM> makeCivilianItemUnequipable = item => {
+        var isCivilian = item?.ItemRosterElement.EquipmentElement.Item?.IsCivilian ?? false;
+        if (isItemApplicableToPerk(item) && !isCivilian) {
+          SetIsCivilianItemMethodInfo.Invoke(item, new object[] { state });
```

Comment:
Is using `.BuildInvoker<Action<SPItemVM,bool>>()` and saving to static readonly possible here?